### PR TITLE
Use `image.registry` value as image registry for all containers in the chart

### PR DESCRIPTION
**Pull request by whites11**
Towards: https://github.com/giantswarm/giantswarm/issues/26243

This PR:

- Use `image.registry` value as image registry for all containers in the chart

### Testing

Description on how cilium can be tested.

- [ ] fresh install works
  - [ ] AWS
  - [ ] Azure
  - [ ] KVM
- [ ] upgrade from previous version works
  - [ ] AWS
  - [ ] Azure
  - [ ] KVM

#### Other testing

Description of features to additionally test for cilium installations.

- [ ] check reconciliation of existing resources after upgrading
- [ ] X still works after upgrade
- [ ] Y is installed correctly

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Make sure `values.yaml` and `values.schema.json` are valid.

### CODE REVIEW

File: helm/cilium/templates/cilium-agent/daemonset.yaml [hunk 1]
```diff
@@ -79,7 +79,7 @@ spec:
       {{- end }}
       containers:
       - name: cilium-agent
-        image: {{ include "cilium.image" .Values.image | quote }}
+        image: "{{ .Values.image.registry }}/{{ include "cilium.image" .Values.image }}"
```
[the remark this answers is not in the document]
Comment (njuettner) [reply helm/cilium/templates/cilium-agent/daemonset.yaml:82]: Ah didn't see that. LGTM

**Review by njuettner — APPROVED** [review]

**Comment by whites11** [comment]
@njuettner i was waiting for Rocket to take a look. Merge at will FWIAC

**Review by kopiczko — APPROVED** [review]

**Comment by whites11** [comment]
Testing one more time in china to ensure it works as expected, then merging

**Comment by whites11** [comment]
looks good, merging

```
$ kubectl --context gs-giraffe-h4u4x-clientcert get po -o yaml -A | grep image:|grep cilium|sort|uniq
      image: giantswarm-registry.cn-shanghai.cr.aliyuncs.com/giantswarm/cilium-operator-generic:v1.13.0
      image: giantswarm-registry.cn-shanghai.cr.aliyuncs.com/giantswarm/cilium:v1.13.0
```

**Pull request merged**